src/drive/web/modules/move/MoveModal.jsx

```diff
@@ -0,0 +1,165 @@
+      console.warn(e)
+      Alerter.error('move error')
+    } finally {
+      onClose()
+    }
+  }
+
+  sortData = data => {
+    const folders = data.filter(
+      entry => entry.type === 'directory' && entry.id !== TRASH_DIR_ID
+    )
+    const files = data.filter(entry => entry.type !== 'directory')
+
+    return folders.concat(files)
+  }
+
+  buildBreadcrumbPath = data =>
+    renamePathNames(
+      getFolderPath({
+        ...data,
+        parent: data.relationships.parent.data
+      }),
+      '',
+      this.context.t
+    )
+
+  render() {
+    const { onClose } = this.props
+    const { client, t } = this.context
+    const { folderId } = this.state
+
+    const contentQuery = client =>
+      client
+        .find('io.cozy.files')
+        .where({ dir_id: folderId })
+        .sortBy({ name: 'asc' })
+
+    const breadcrumbQuery = client => client.get('io.cozy.files', folderId)
+
+    return (
+      <Modal size={'xlarge'} closable={false} overflowHidden mobileFullscreen>
+        <Query query={breadcrumbQuery} key={`breadcrumb-${folderId}`}>
+          {({ data, fetchStatus }) => {
+            return fetchStatus === 'loaded' ? (
+              <MoveTopbar
+                navigateTo={this.navigateTo}
+                path={this.buildBreadcrumbPath(data)}
+              />
+            ) : (
+              false
+            )
+          }}
+        </Query>
+        <Query query={contentQuery} key={`content-${folderId}`}>
+          {({ data, fetchStatus }) => {
+            return (
+              <div className={fileListStyles['fil-content-table']} role="table">
+                <MobileFileListHeader canSort={false} />
+                <FileListHeader canSort={false} />
+                <div className={fileListStyles['fil-content-body']}>
+                  {/*Missing FileListBody providing the loading state, the empty state and the add folder component */}
+                  <div>
+                    {/*Missing FileListRows providing the load more  */}
+                    {this.sortData(data).map(file => (
+                      <File
+                        key={file.id}
+                        attributes={file}
+                        displayedFolder={null}
+                        actions={null}
+                        isRenaming={false}
+                        onFolderOpen={id =>
+                          this.navigateTo(data.find(f => f.id === id))
+                        }
+                        onFileOpen={null}
+                        withSelectionCheckbox={false}
+                        withFilePath={false}
+                        withSharedBadge={true}
+                      />
+                    ))}
+                  </div>
+                </div>
+              </div>
+            )
+          }}
+        </Query>
+        <div>
+          <Button theme="secondary" onClick={onClose}>
+            Annuler
+          </Button>
+          <Button onClick={this.moveEntries}>Déplacer</Button>
```

Comment:
Everytime we'll render the `MoveModal`, we'll trigger an update of this button because everytime we create an instance of `moveEntries`

If we had, in the constructor : `this.moveEntries = this.moveEntries.bind(this)`, we'll create an instance of this method only once.  And if the `Button` is smart enough, it will not be rendered several times :)

Reply:
Never mind, didn't see the arrow function :(

Reply:
So our guidelines currently [recommend the opposite](https://github.com/cozy/cozy-guidelines#binding-event-handlers), but it's debatable. We can open an issue there, you're not the first one to raise this issue.

That being said, are you sure it will re-render the button? `this.moveEntries` is a reference to our function, and the reference doesn't change, even when doing a new render pass.